packages/@lwc/ssr-compiler/src/compile-js/errors.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { generateErrorMessage, type LWCErrorInfo } from '@lwc/errors';
+
+// This type extracts the arguments in a string. Example: "Error {0} {1}" -> [string, string]
+type ExtractArguments<T extends string> = T extends `${string}{${number}}${infer R}`
+    ? [string, ...ExtractArguments<R>]
+    : [];
```

Comment:
This fails for `SINGLE_DECORATOR_ON_SETTER_GETTER_PAIR` (`'@api get {0} and @api set {0} detected in class declaration. Only one of the two needs to be decorated with @api.'`) because of the duplicate `{0}`.

Reply:
`````suggestion
type ExtractArguments<
    T extends string,
    Numbers extends number = never,
    Args extends string[] = [],
> = T extends `${string}{${infer N extends number}}${infer R}`
    ? N extends Numbers // Is `N` in the union of seen numbers?
        ? ExtractArguments<R, Numbers, Args> // new `N`, add an argument
        : ExtractArguments<R, N | Numbers, [string, ...Args]> // `N` already accounted for
    : Args; // No `N` found, nothing more to check
```

Reply:
When I grow up I want to be a typescript ninja just like you! 🤯